packages/libp2p/test/connection-manager/direct.spec.ts

```diff
@@ -402,6 +403,57 @@ describe('libp2p.dialer (direct, WebSockets)', () => {
     await libp2p.stop()
   })
 
+  it('should not run identify automatically after connecting', async () => {
+    libp2p = await createLibp2p({
+      peerId,
+      transports: [
+        webSockets({
+          filter: filters.all
+        })
+      ],
+      streamMuxers: [
+        yamux()
+      ],
+      connectionEncryption: [
+        plaintext()
+      ],
+      services: {
+        identify: identifyService({
+          protocolPrefix: 'ipfs',
+          agentVersion: AGENT_VERSION,
+          timeout: 60000,
+          maxInboundStreams: 1,
+          maxOutboundStreams: 1,
+          maxPushIncomingStreams: 1,
+          maxPushOutgoingStreams: 1,
+          maxObservedAddresses: 10,
+          maxIdentifyMessageSize: 8192,
+          runOnConnectionOpen: false
+        })
+      },
+      connectionGater: mockConnectionGater()
+    })
+
+    if (libp2p.services.identify == null) {
+      throw new Error('Identify service missing')
+    }
+
+    const identifySpy = sinon.spy(libp2p.services.identify as DefaultIdentifyService, 'identify')
```

Comment:
Since `identify` is in the interface now, do we still need the cast?

Reply:
Unfortunately yes, I think for similar limitations raised in https://github.com/libp2p/js-libp2p/pull/1762 the compiler doesn't detect the `IdentifyService` interface.

Reply:
I'm not sure this is correct - the `libp2p` variable is declared as `let libp2p: Libp2p<{ identify: unknown }>`.

If it was `let libp2p: Libp2p<{ identify: IndentifyService }>` the cast should be able to be removed.